convert2rhel/pkghandler.py

```diff
@@ -36,6 +36,28 @@
 _VERSIONLOCK_FILE_PATH = "/etc/yum/pluginconf.d/versionlock.list"  # This file is used by the dnf plugin as well
 versionlock_file = utils.RestorableFile(_VERSIONLOCK_FILE_PATH)  # pylint: disable=C0103
 
+#
+# Regular expressions used to find package names in yum output
+#
+
+# This regex finds package NEVRs + arch (name epoch version release and
+# architechture) in a string.  It separates the epoch from the rest of the
+# NEVR, leaving us to parse the name from the remaining NVR.
+# Note that the regex requires at least two dashes but the NVR can contain
+# more than that.  For instance: gcc-c++-4.8.5-44.0.3.el7.x86_64
+PKG_NEVR = r"\b([0-9]+:)?(\S+-\S+-\S+)\b"
```

Comment:
Since first group isn't used should we make it non-capturing?
```suggestion
PKG_NEVR = r"\b(?:[0-9]+:)?(\S+-\S+-\S+)\b"
```

Reply:
This RegEx also changes the group from the system name to the whole name. Currently we get a group `systemd-python` but with this change we'd get `systemd-python-219-78.0.7.el7_9.3.x86_64`

Reply:
We could also do this so we do not need to do split later
```suggestion
PKG_NEVR = r"\b(?:[0-9]+:)?(\S+)-(\S+)-(\S+)\b"
```

Reply:
Yeah, I think that would work and then the function can be reduced to doing less work.  I have time this morning so let me try to whip that up.

Reply:
Updated the code to use almost this RegEx.  I modified it slightly so that it will also capture the epoch in case we need to retrieve the epoch, version, and release in the future.